messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+                                                           + "where dl.processingGroup=:processingGroup",
+                                                   Long.class
+                                           )
+                                           .setParameter("processingGroup", processingGroup)
+                                           .getSingleResult());
+    }
+
+    /**
+     * Fetches the next maximum index for a sequence that should be used when inserting a new item into the databse for
+     * this {@code sequence}.
+     *
+     * @param sequence The identifier of the queue to fetch the next index for.
+     * @return The next sequence index.
+     */
+    private Long getNextIndexForSequence(String sequence) {
+        Long maxIndex = getMaxIndexForSequence(sequence);
+        if (maxIndex == null) {
+            return 0L;
+        }
+        return maxIndex + 1;
+    }
+
+    /**
+     * Fetches the current maximum index for a queue identifier. Messages which are enqueued next should have an index
+     * higher than the one returned. If the query returns null it indicates that the queue is empty.
+     *
+     * @param sequence The identifier of the queue to check the index for.
+     * @return The current maximum index, or null if not present.
+     */
+    private Long getMaxIndexForSequence(String sequence) {
+        return transactionManager.fetchInTransaction(() -> {
+            try {
+                return entityManager().createQuery(
+                                              "SELECT max(dl.index) FROM DeadLetterEntry dl where dl.processingGroup=:processingGroup and dl.sequence=:sequence",
+                                              Long.class
+                                      )
+                                      .setParameter("sequence", sequence)
+                                      .setParameter("processingGroup", processingGroup)
+                                      .getSingleResult();
+            } catch (NoResultException e) {
+                // Expected, queue is empty. Return null.
+                return null;
+            }
+        });
+    }
+
+    /**
+     * Converts the given sequence identifier to a String.
+     */
+    private String toIdentifier(Object sequenceIdentifier) {
+        if (sequenceIdentifier instanceof String) {
+            return (String) sequenceIdentifier;
+        }
+        return Integer.toString(sequenceIdentifier.hashCode());
+    }
+
+
+    /**
+     * Builder class to instantiate an {@link JpaSequencedDeadLetterQueue}.
+     * <p>
+     * The maximum number of unique sequences defaults to {@code 1024}, the maximum amount of dead letters inside an
+     * unique sequence to {@code 1024}, the claim duration defaults to {@code 30} seconds and the converters default to
+     * containing a single {@link EventMessageDeadLetterJpaConverter}.
```

Comment:
I would expect the JavaDoc to state which fields the user is inclined to configure.